src/screens/WeatherScreen.js

```diff
@@ -52,26 +54,49 @@ export const WeatherScreen = () => {
     fetchPolicy === 'network-only'
       ? { fetchPolicy, pollInterval: POLL_INTERVALS.WEATHER }
       : { fetchPolicy };
-  const { data, loading } = useQuery(getQuery(QUERY_TYPES.WEATHER_MAP), queryVariables);
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.WEATHER_MAP), queryVariables);
 
   useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.WEATHER);
 
-  if (loading) {
+  if (loading && !data?.weatherMap) {
```

Comment:
is `loading` not enough. what about if loading and there is data? we would see the previous data until the data changes, right?

Reply:
your seceond/third question is kind of the answer to the first one.
`loading` would be enough for it being stable, but if we have previous data, we would like to show it together with an activity indicator on top.